tests/infra/clients.py

```diff
@@ -415,12 +418,36 @@ def __init__(
         self.key = key
         self.ca = ca
         self.request_timeout = request_timeout
+        self.ws = None
 
     def request(self, request):
-        # pylint: disable=unused-variable
-        ws = create_connection(
-            f"wss://{self.host}:{self.port}",
-            sslopt={"certfile": self.cert, "keyfile": self.key, "ca_certs": self.ca},
+        if not self.ws:
+            LOG.info("Creating WSS connection")
+            self.ws = websocket.create_connection(
+                f"wss://{self.host}:{self.port}",
+                sslopt={
+                    "certfile": self.cert,
+                    "keyfile": self.key,
+                    "ca_certs": self.ca,
+                },
+            )
+        payload = json.dumps(request.params).encode()
+
+        builder = flatbuffers.Builder(1)
+        path = builder.CreateString("/" + request.method)
+        InHeader.InHeaderStart(builder)
+        InHeader.InHeaderAddPath(builder, path)
+        header = InHeader.InHeaderEnd(builder)
+        builder.Finish(header)
+        h = builder.Output()
+        # FIN, no RSV, BIN, UNMASKED every time, because it's all we support right now
+        frame = websocket.ABNF(1, 0, 0, 0, websocket.ABNF.OPCODE_BINARY, 0, h + payload)
+        self.ws.send_frame(frame)
+        out = self.ws.recv_frame().data
+        ohs = 20  # Calculate, somehow
```

Comment:
In C++, the Verifier can do that, so long as the right define is set. No such luck in Python!

Slightly awkward, but not impossible to monkey patch.

Reply:
Actually this seems to be genuinely difficult...

Reply:
Going to skip that for now in favour of trying to get smallbank running asap.